iroh-net/src/endpoint.rs

```diff
@@ -450,8 +450,13 @@ impl Endpoint {
     /// The `alpn`, or application-level protocol identifier, is also required. The remote
     /// endpoint must support this `alpn`, otherwise the connection attempt will fail with
     /// an error.
-    #[instrument(skip_all, fields(me = %self.node_id().fmt_short(), remote = %node_addr.node_id.fmt_short(), alpn = ?String::from_utf8_lossy(alpn)))]
-    pub async fn connect(&self, node_addr: NodeAddr, alpn: &[u8]) -> Result<quinn::Connection> {
+    #[instrument(skip_all, fields(me = %self.node_id().fmt_short(), alpn = ?String::from_utf8_lossy(alpn)))]
```

Comment:
Yeah, it is clear you want it. Otherwise you wouldn't have added it in the first place. But the only way I can see to keep it in is to do as described and do a private _impl fn where the addr is already converted to a NodeAddr.

Will do that if you don't have a better idea.

Reply:
Pushed an update. You can just add something to the current span it seems...